js/src/logger.ts

```diff
@@ -748,6 +748,64 @@ function logFeedbackImpl(
   }
 }
 
+function updateSpanImpl(
+  state: BraintrustState,
+  parentObjectType: SpanObjectTypeV2,
+  parentObjectId: LazyValue<string>,
+  id: string,
+  event: Partial<ExperimentEvent>,
+): void {
+  const updateEvent = validateAndSanitizeExperimentLogPartialArgs({
+    id,
+    ...event,
+  });
+
+  const parentIds = async () =>
+    new SpanComponentsV2({
+      objectType: parentObjectType,
+      objectId: await parentObjectId.get(),
+    }).objectIdFields();
+
+  const record = new LazyValue(async () => ({
+    id,
+    ...updateEvent,
+    ...(await parentIds()),
+    [IS_MERGE_FIELD]: true,
+  }));
+  state.bgLogger().log([record]);
+}
+
+/**
+ * Update a span using the output of `span.export()`. It is important that you only resume updating
+ * to a span once the original span has been fully written and flushed, since otherwise updates to
+ * the span may conflict with the original span.
+ *
+ * @param exported The output of `span.export()`.
+ * @param event The event data to update the span with. See `Experiment.log` for a full list of valid fields.
+ * @param state (optional) Login state to use. If not provided, the global state will be used.
+ */
+export function updateSpan({
+  exported,
+  state,
+  ...event
+}: { exported: string } & Omit<Partial<ExperimentEvent>, "id"> &
```

Comment:
Should we also have `Omit<Partial<ExperimentEvent>, "id">` in the `updateSpanImpl` type signature for extra safety?

Reply:
yep.